string node_name
---
uint8 level
bool is_valid